edgexfoundry/device-modbus-go
# refactor(snap): edgex-snap-hooks related upgrade

This PR upgrades snap hooks to [2.4.1 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.4.1)with new function and refactor:
- Combine install hook and configure hook
- Use library function for copying directory
- Remove export bash command to process valid .env files
- Use log package for logging
- Use library function for autostart processing

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-modbus-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?) [modbus test suite](https://github.com/canonical/edgex-snap-testing/tree/main/test/suites/device-modbus) passed with `FULL_CONFIG_TEST=true`
  <link to docs PR>

## Testing Instructions

1. Install and setup:
```
sudo snap set edgex-device-modbus app-options=true
sudo snap set edgex-device-modbus config.writable-loglevel=DEBUG
sudo snap set edgex-device-modbus config.service-startupmsg="#test start up message#"
sudo snap start edgex-device-modbus
```
2. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-device-modbus/current/config/device-modbus/res/device-modbus.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
3. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs -f edgex-device-modbus

msg="#test start up message#"
level=DEBUG msg="xxx"
```
4. Validate all config files have been copied in right place:
```
tree /var/snap/edgex-device-modbus/current/config
.
└── device-modbus
    └── res
        ├── configuration.toml
        ├── devices
        │   └── modbus.test.devices.toml
        └── profiles
            ├── modbus.test.device.profile.yml
            └── modbus.test.device.string.profile.yml

4 directories, 5 files

```
5. Validate autostart via snap option works:
```
$ sudo snap set edgex-device-modbus autostart=true
$ snap services edgex-device-modbus
Service                            Startup  Current  Notes
edgex-device-modbus.device-modbus  enabled  active   -

$ sudo snap set edgex-device-modbus autostart=false
$ snap services edgex-device-modbus
Service                            Startup   Current   Notes
edgex-device-modbus.device-modbus  disabled  inactive  -
```

## New Dependency Instructions (If applicable)
